### ISSUE TITLE
Some suggestions/ideas

### ISSUE THREAD

**Issue by zeenk** [Xruptor/BagSync#326]
## Quick access button on BagSync windows
(*Excuse my poor paint skills*) 😄 
Would be nice to have a quick access button on all BagSync windows (or option for it at least) so you don't have to go back to minimap every time if you want to check something else real quick.
Or you could place the button on the right side next to the close button if you think that would look better.
![Main](https://github.com/Xruptor/BagSync/assets/68170439/b9a7d139-27d4-4d21-8ba6-5db68c4c2a33)

## Currency sort/search
Options to sort the currency list like the default tab instead of alphabetical so the relevant currencies are at the top, maybe also add a search bar to this window? quite a lot of currencies has been added over the years.. ^^
Would also be nice to just have currency support in the search window so you don't even need to open the currency window and can just search for them directly in the search window.
![2024-02-19_04-00-13](https://github.com/Xruptor/BagSync/assets/68170439/37a56fb4-dcd6-4968-b498-ab2ae9d88761)
![2024-02-19_04-19-50](https://github.com/Xruptor/BagSync/assets/68170439/44e65c18-07ad-4db2-8216-501579b34749)

## Option to show current realm in tooltips
I use these settings for tooltips, but if you decide to sort things by total instead of alphabetical it looks a bit silly in the tooltips so would be nice to have an extra option to also show current realm so everything lines up better in the tooltip😄 
![2024-02-19_04-38-08](https://github.com/Xruptor/BagSync/assets/68170439/bb44cd7f-200d-4c95-9555-ce746a17f847)
![2024-02-19_04-38-20](https://github.com/Xruptor/BagSync/assets/68170439/8f3dbc6a-1d7f-4a3e-b863-05f22acf1bb1)
![2024-02-19_03-52-13](https://github.com/Xruptor/BagSync/assets/68170439/ad1fe4a4-d178-423a-a0f0-1a8731d67035)

## Improved search
Just a minor thing but would be nice if the search window could match stuff in real-time instead of only after you have hit enter ^^

https://github.com/Xruptor/BagSync/assets/68170439/640b0b98-8e24-48c1-a95a-e4211fe0d9a4

Sorry for the lengthy post, hopefully you at least like something from my suggestions 😄

**Comment by Xruptor** [comment]
I apologize for the delay!  I've been out of town the last few days.  Let me go over and review these suggestions.  😄

**Comment by Xruptor** [comment]
> ## Quick access button on BagSync windows
> (_Excuse my poor paint skills_) 😄 Would be nice to have a quick access button on all BagSync windows (or option for it at least) so you don't have to go back to minimap every time if you want to check something else real quick. Or you could place the button on the right side next to the close button if you think that would look better. ![Main](https://private-user-images.githubusercontent.com/68170439/305777808-b9a7d139-27d4-4d21-8ba6-5db68c4c2a33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0NjU4MzMsIm5iZiI6MTcwODQ2NTUzMywicGF0aCI6Ii82ODE3MDQzOS8zMDU3Nzc4MDgtYjlhN2QxMzktMjdkNC00ZDIxLThiYTYtNWRiNjhjNGMyYTMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIwVDIxNDUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjZDAzYjM3NzA0MWM1MGMwYTJhNDNjMTg5ZWU0NTZkMTJkODEyMzlkMjBmNzYyODQ0NTcwZGQ0OWZiMTc3YmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.47BQclGZdq9Hu3wf9DLvyp9TlV1E8cTmpyutQFl3CAo)
> 

I think this is a great idea.  I'll look into how I can add it without causing issues with the frame levels.  Dropdowns can be quite tricky sometimes believe it or not.  I'll see if I can implement this.

> ## Currency sort/search
> Options to sort the currency list like the default tab instead of alphabetical so the relevant currencies are at the top, maybe also add a search bar to this window? quite a lot of currencies has been added over the years.. ^^ Would also be nice to just have currency support in the search window so you don't even need to open the currency window and can just search for them directly in the search window. ![2024-02-19_04-00-13](https://private-user-images.githubusercontent.com/68170439/305779494-37a56fb4-dcd6-4968-b498-ab2ae9d88761.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0NjU4MzMsIm5iZiI6MTcwODQ2NTUzMywicGF0aCI6Ii82ODE3MDQzOS8zMDU3Nzk0OTQtMzdhNTZmYjQtZGNkNi00OTY4LWI0OTgtYWIyYWU5ZDg4NzYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIwVDIxNDUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjODg4NDk1ZTVkNDgwYTIyYmE3ZDcwMmQ2ZTI4MDk3NjkwN2VmYWMyMWY5NWZjYTYyNTIyYzBjOTViY2VhNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QU_dVU4ECybKddUYXqp3OF8XVVIrmXfvgOcGxMhT7Qw) ![2024-02-19_04-19-50](https://private-user-images.githubusercontent.com/68170439/305779569-44e65c18-07ad-4db2-8216-501579b34749.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0NjU4MzMsIm5iZiI6MTcwODQ2NTUzMywicGF0aCI6Ii82ODE3MDQzOS8zMDU3Nzk1NjktNDRlNjVjMTgtMDdhZC00ZGIyLTgyMTYtNTAxNTc5YjM0NzQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIwVDIxNDUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhNTA1ODY3YmM5ZmRlM2U0MGVhY2FjZTk4OTE5OWViZDkzOTUwZjIyMzE3MTI0M2RmMTI1ODcyY2ExMDQ2YjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8DO-obX8MfJZdK4m2Ddm_vK-_gMfNLqfTkoCnp_Sbcw)
> 

Back in the day I originally had it working like this but changed it to be alphabetical instead.  If I add any type of sorting it's going to have to be done as an option by current expansion.  I really don't want to add a search feature as that would be additional code I don't want to include in that frame at the moment.


> ## Option to show current realm in tooltips
> I use these settings for tooltips, but if you decide to sort things by total instead of alphabetical it looks a bit silly in the tooltips so would be nice to have an extra option to also show current realm so everything lines up better in the tooltip😄 ![2024-02-19_04-38-08](https://private-user-images.githubusercontent.com/68170439/305782017-bb44cd7f-200d-4c95-9555-ce746a17f847.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0NjU4MzMsIm5iZiI6MTcwODQ2NTUzMywicGF0aCI6Ii82ODE3MDQzOS8zMDU3ODIwMTctYmI0NGNkN2YtMjAwZC00Yzk1LTk1NTUtY2U3NDZhMTdmODQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIwVDIxNDUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiOTg4NGE5ZjhlNWM4MjQ0N2U3ZTFhNjBmNTEyYTQ5MDFlN2NhMGRkMTQ5MmUxNTc3N2Y3YzFjNDY2NmUwMzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uKNNBnfBNjtWZpQ6Kk7K42H9CLxfySVYHdt42yIE19s) ![2024-02-19_04-38-20](https://private-user-images.githubusercontent.com/68170439/305786246-8f3dbc6a-1d7f-4a3e-b863-05f22acf1bb1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0NjU4MzMsIm5iZiI6MTcwODQ2NTUzMywicGF0aCI6Ii82ODE3MDQzOS8zMDU3ODYyNDYtOGYzZGJjNmEtMWQ3Zi00YTNlLWI4NjMtMDVmMjJhY2YxYmIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIwVDIxNDUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkYjdhY2JkNDZjZDUwMTZjOTA5MjFmMzNlN2RhYzI1MjYzZDFjNjdkZTMxZDM1NGE2ZWZjODc1YmQwODRmNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.V5xUNOuu9WcpmKhkBngl9uXXfYq3-SPioaBVFlYoh-M) ![2024-02-19_03-52-13](https://private-user-images.githubusercontent.com/68170439/305781809-ad1fe4a4-d178-423a-a0f0-1a8731d67035.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0NjU4MzMsIm5iZiI6MTcwODQ2NTUzMywicGF0aCI6Ii82ODE3MDQzOS8zMDU3ODE4MDktYWQxZmU0YTQtZDE3OC00MjNhLWEwZjAtMWE4NzMxZDY3MDM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIwVDIxNDUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwOTU2N2JkYWExNWU2YTNjMDBlZTM2NzJjODgwNTQ5NjcwOTk1OGI3YzRlMzMzODdmZDE3N2FhNGQyODg3ZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VtdyATqyUHJ0XV8CHpYdG8FhBDtGnekKpbB4cUrR3fk)
> 

This actually makes sense to me and should be an option.  I'll look into adding this and just allowing folks to enable realm names for their current realms in general.

> ## Improved search
> Just a minor thing but would be nice if the search window could match stuff in real-time instead of only after you have hit enter ^^
> 

I was originally going to do this but I decided against it due to a technical situation with how the cache is generated and the interactions with the advanced search system.  I can review it again and see if it can be done, but if I recall it added a whole mess of bugs with focus search and advanced search settings.  Basically with each key stroke it parses the entire database each time, which on some computers can cause extensive lag or delays depending on the size of the database.  In addition the advanced search routine puts that into blocks before parsing.  Like I said I'll have to look into it, but I highly doubt I can add this without causing some significant impact to performance.

>  305661443-74d5ce0b-7f77-4a21-b88c-9127231ae9bc.mp4 
> Sorry for the lengthy post, hopefully you at least like something from my suggestions 😄

No problem!  I love suggestions and I usually try my best to implement stuff people want as long as it's reasonable.

**Comment by zeenk** [comment]
Glad you liked some of them 😄 But yeah if the real-time search will cause lag just skip that, it's no big deal 👍

**Comment by killerbonzai** [comment]
I would like (if its ok) to add to the "Option to show current realm in tooltips".
If you are in that area anyway :)

I play hardcore classic trying to do a couple of characters at the same time, but only selffound items for all my characters.
It would be super great if I could just say "Show only this character" or maybe only show certain characters (to filter out the dead ones or if I decide to do both selffound and not at the same time).

I hope this makes sense in the same way it does in my head :)
It seemed harder to explain here than I first thought.

TL;DR:
An option to "Show only this character" 
Or "Show only certain characters" -> mark them in a list

**Comment by Xruptor** [comment]
> I would like (if its ok) to add to the "Option to show current realm in tooltips". If you are in that area anyway :)
> 
> I play hardcore classic trying to do a couple of characters at the same time, but only selffound items for all my characters. It would be super great if I could just say "Show only this character" or maybe only show certain characters (to filter out the dead ones or if I decide to do both selffound and not at the same time).
> 
> I hope this makes sense in the same way it does in my head :) It seemed harder to explain here than I first thought.
> 
> TL;DR: An option to "Show only this character" Or "Show only certain characters" -> mark them in a list

So you only want to show information for the currently logged in character? That's it right?

**Comment by killerbonzai** [comment]
Yes.
That would be nice for my self-found characters :)

**Comment by zeenk** [comment]
Thanks! I see you added the quick access button on search frame and realm options, looks better now when showing all chars and sorting by totals 👍 (I changed the color to blue for all realms, I prefer it that way :P)

![2024-03-03_22-20-04](https://github.com/Xruptor/BagSync/assets/68170439/9fd6f2a1-e9ba-40f9-b874-764ebc918884)

**Comment by zeenk** [comment]
No worries, it's fine I understand 🙂
